src/vm/eventreporter.cpp

```diff
@@ -718,23 +726,37 @@ void DoReportForUnhandledException(PEXCEPTION_POINTERS pExceptionInfo)
                 // look that up and if it is not empty, add those details to the EventReporter so that they get written to the
                 // event log as well.
                 //
-                // We can also be here when in non-DefaultDomain an exception goes unhandled on a pure managed thread. In such a case,
-                // we wont have CrossAppDomainMarshaledException instance but the original exception object that will be used to extract
-                //  the stack trace from.
                 if (IsException(gc.throwable->GetMethodTable()))
                 {
-                    SmallStackSString wordAt;
-                    if (!wordAt.LoadResource(CCompRC::Optional, IDS_ER_WORDAT))
+                    StackSString result;
+                    // Assume we're calling Exception.InternalToString() ...
+                    BinderMethodID sigID = METHOD__EXCEPTION__INTERNAL_TO_STRING;
+
+                    // Get the MethodDesc on which we'll call.
+                    MethodDescCallSite toString(sigID, &(gc.throwable));
```

Comment:
This may cause trouble. If you look at InternalUnhandledExceptionFilter_Worker, after it calls this method it then calls DefaultCatchHandler. DefaultCatchHandler also invokes Exception.ToString() in some cases, but first it does a bunch of checks to identify cases where it isn't safe to make the managed call.
https://github.com/dotnet/coreclr/blob/master/src/vm/excep.cpp#L5674

Some of those issues appear irrelevant for CoreCLR such as multiple AppDomains or ThreadAbort, but others such as OutOfMemory, StackOverflow, and CanCallManagedCode() are potentially all relevant. We should also investigate what error reporting tests exist which hopefully probe some of these cases. I suspect the tests exist in the desktop .Net tree but haven't yet been ported.